engine/crates/integration-tests/tests/federation/hooks/on_subgraph_request.rs

```diff
@@ -0,0 +1,245 @@
+use engine_v2::Engine;
+use graphql_mocks::{EchoSchema, FakeGithubSchema, MockGraphQlServer};
+use http::HeaderMap;
+use integration_tests::{federation::EngineV2Ext, runtime};
+use runtime::{
+    error::{PartialErrorCode, PartialGraphqlError},
+    hooks::{DynHookContext, DynHooks},
+};
+use runtime_local::HooksWasiConfig;
+use url::Url;
+
+#[test]
+fn wasi() {
```

Comment:
We should leave a note for people to compile the hooks before running the tests. I have a hunch this might be annoying. If you just `cargo test` here this will fail if you don't compile the hooks examples first.

Reply:
oh right, good catch. Was thinking of CI, not devs